lib/Sema/PlaygroundTransform.cpp

```diff
@@ -890,25 +892,33 @@ class Instrumenter : InstrumenterBase {
         ArgumentList::forImplicitUnlabeled(Context, ArgsWithSourceRange);
     ApplyExpr *LoggerCall =
         CallExpr::createImplicit(Context, LoggerRef, ArgList);
-    Added<ApplyExpr *> AddedLogger(LoggerCall);
 
-    if (!doTypeCheck(Context, TypeCheckDC, AddedLogger)) {
+    Expr *E = LoggerCall;
+    // Type-check the newly created logger call. Note that the type-checker is
+    // free to change the expression type, so type-checking is performed
+    // before wrapping in Added<>.
+    if (!doTypeCheckExpr(Context, TypeCheckDC, E)) {
+      return nullptr;
+    }
+
+    Added<Expr *> AddedLogger(E);
+    if (*AddedLogger == nullptr) {
```

Comment:
Just curious if you ever saw this be `nullptr` here.  By my reading it should not be possible for `doTypeCheckExpr()` to have returned `true` but then the expression to have been set to `nullptr` (and it looks as if `Added<T>` just wraps whatever is given).

This isn't a request for change — it's great to be defensive here (and in fact I think the playground transform should be _a lot_ more defensive in general, so that failures result in missing logger calls rather than compiler crashes!

Just curious as I read through the code in detail and tested out the changes locally.

Reply:
Good call out. You're right, this check should be unnecessary.  Originally, `doTypeCheck()` was setting `Added.Contents` to null by a failed dyn_cast, which was the cause of the crash. I had added this check to fail gracefully in that case, before applying the full fix.  This check is no longer needed, as the Added constructor shouldn't fail to wrap the expression, and `doTypeCheckExpr()` will return false if expression is null after type checking.

Reply:
Great, thanks for clarifying and confirming.  To be clear, I don't think it hurts, just in case we ever do end up with `nullptr` here.